### ISSUE TITLE
cron.php: Call to a member function addRecord() on null in /usr/share/glpi/inc/toolbox.class.php:435

### ISSUE THREAD

**Issue by sbraz** [glpi-project/glpi#6690]
**Describe the bug**
GLPI's cron script fails with the following message:
```
PHP Fatal error:  Uncaught Error: Call to a member function addRecord() on null in /usr/share/glpi/inc/toolbox.class.php:434
Stack trace:
#0 /usr/share/glpi/inc/toolbox.class.php(468): Toolbox::log(NULL, 400, Array)
#1 /usr/share/glpi/inc/config.class.php(3110): Toolbox::logError('Cache directory...')
#2 /usr/share/glpi/inc/config.php(45): Config::getCache('cache_db')
#3 /usr/share/glpi/inc/includes.php(48): include_once('/usr/share/glpi...')
#4 /usr/share/glpi/front/cron.php(38): include('/usr/share/glpi...')
#5 {main}
  thrown in /usr/share/glpi/inc/toolbox.class.php on line 434
```

FWIW the message that it should log is the following (probably unrelated) to the `$PHPLOGGER` being `NULL`:
```
Config::getCache() in /usr/share/glpi/inc/config.class.php line 3109
Cache directory '/var/lib/glpi/files/_cache/cache_db' not writable 
```

**To reproduce**
```
sudo -u apache /usr/bin/php /usr/share/glpi/front/cron.php
```

**Your GLPI setup (you can find it in Setup > General menu, System tab)**
```
GLPI 9.4.4 (/glpi => /usr/share/glpi)
Installation mode: RPM
```

Someone else has the same problem [here](https://forum.glpi-project.org/viewtopic.php?id=196374).

**Comment by cedric-anne** [comment]
Hi,

This has been fixed by #6446 .

Regards

**Comment by sbraz** [comment]
Looks like the issue is indeed linked to the `apache` user (with which the cron task runs on on CentOS with the RPM install) not having write permissions on `/var/lib/glpi/files/_cache/cache_db`:
```
drwxr-sr-x. 6 root apache 254 Dec  9 16:06 /var/lib/glpi/files/_cache/cache_db
```

After deleting the `cache_db` directoryu, I saw a bunch of other errors with other directories that I also removed. Overall, running the following commands fixed the problem:
```
rm -rf /var/lib/glpi/files/_cache/cache_db/
rm -rf /var/lib/glpi/files/_cache/cache_trans/
rm -f /var/lib/glpi/files/_cache/glpi_cache_db_9.4.4.json
```


@cedric-anne thanks for the reply, I'm still leaving my comment in case it helps anybody else if they also messed their cache permissions.
